src/controller/CommissionerDiscoveryController.cpp

```diff
@@ -94,24 +120,73 @@ void CommissionerDiscoveryController::OnUserDirectedCommissioningRequest(UDCClie
     mReady = false;
     Platform::CopyString(mCurrentInstance, state.GetInstanceName());
     mPendingConsent = true;
-    char rotatingDeviceIdHexBuffer[RotatingDeviceId::kHexMaxLength];
-    Encoding::BytesToUppercaseHexString(state.GetRotatingId(), state.GetRotatingIdLength(), rotatingDeviceIdHexBuffer,
-                                        RotatingDeviceId::kHexMaxLength);
+    char rotatingIdString[Dnssd::kMaxRotatingIdLen * 2 + 1];
+    Encoding::BytesToUppercaseHexString(state.GetRotatingId(), state.GetRotatingIdLength(), rotatingIdString,
+                                        sizeof(rotatingIdString));
 
     ChipLogDetail(Controller,
                   "------PROMPT USER: %s is requesting permission to cast to this TV, approve? [" ChipLogFormatMEI
                   "," ChipLogFormatMEI ",%s,%s]",
                   state.GetDeviceName(), ChipLogValueMEI(state.GetVendorId()), ChipLogValueMEI(state.GetProductId()),
-                  state.GetInstanceName(), rotatingDeviceIdHexBuffer);
+                  state.GetInstanceName(), rotatingIdString);
     if (mUserPrompter != nullptr)
     {
         mUserPrompter->PromptForCommissionOKPermission(state.GetVendorId(), state.GetProductId(), state.GetDeviceName());
     }
     ChipLogDetail(Controller, "------Via Shell Enter: controller ux ok|cancel");
 }
 
+/// Callback for getting execution into the main chip thread
+void CallbackOk(System::Layer * aSystemLayer, void * aAppState)
+{
+    ChipLogError(AppServer, "UX Ok: now on main thread");
+    GetCommissionerDiscoveryController()->InternalOk();
+}
+
+/// Callback for getting execution out of the main chip thread
+void CallbackExternalWorkerOk(System::Layer * aSystemLayer, void * aAppState)
+{
+    ChipLogError(AppServer, "UX Ok: out of main thread");
+
+    UDCClientState * client = static_cast<UDCClientState *>(aAppState);
```

Comment:
Does something guarantee that client state does not get overwritten while the callback was pending, or while it's now executed?  Last I checked, FindUDCClientState would happily return you a pointer to a state just based on time elapsed, so if things take too long somewhere someone can have stomped on the client state, no?

Reply:
reverted